src/include/interval_set.h

```diff
@@ -46,6 +46,14 @@ class interval_set {
   class iterator : public std::iterator <std::forward_iterator_tag, T>
   {
     public:
+        using Map = C<T, T>;
+        using value_type = typename Map::value_type;
+        using offset_type = T;
+        using length_type = T;
+        using reference = value_type&;
+        using const_reference = const value_type&;
+        using size_type = typename Map::size_type;
+
```

Comment:
Nit: I think you would technically need `using iterator_category = std::forward_iterator_tag` as well (`difference_type`, `value_type`, `pointer`, `reference`, and `iterator_category` are the 5 required).

Reply:
i think these types are defined by `std::iterator<>` already.

Reply:
@wjwithagen  could you address Jason's comment?